pkg/storage/couchdb/couchdbstore.go

```diff
@@ -505,6 +527,58 @@ func (c *CouchDBStore) getStoredValueFromRawDoc(rawDoc map[string]interface{}, k
 	return strippedJSON, nil
 }
 
+func (c *CouchDBStore) getStoredValuesFromRawDocs(rawDocs []map[string]interface{}, keys []string) ([][]byte, error) {
+	storedValues := make([][]byte, len(keys))
+
+	for i, rawDoc := range rawDocs {
+		if rawDoc == nil {
+			return nil, fmt.Errorf(getBulkKeyNotFound, keys[i], storage.ErrValueNotFound)
+		}
+
+		// CouchDB still returns a raw document is the key has been deleted, so if this is a "deleted" raw document
+		// then we need to return the "value not found" error in order to maintain consistent behaviour with
+		// other storage implementations.
+		isDeleted, containsIsDeleted := rawDoc["_deleted"]
+		if containsIsDeleted {
+			isDeletedBool, ok := isDeleted.(bool)
+			if !ok {
+				return nil, errFailToAssertDeletedAsBool
+			}
+
+			if isDeletedBool {
+				return nil, fmt.Errorf(getBulkKeyNotFound, keys[i], storage.ErrValueNotFound)
```

Comment:
Instead of returning an error, I would suggest that you set this entry to nil. The caller will realize that one of the documents he's requested is nil and therefore deleted or not found.

Reply:
Just talked with @bstasyszyn - the use case for this method is within some internal document processing in the EDV server, and in that case if any retrievals fail then the whole thing has to be considered a failure.